### PR TITLE
CDPS-774: Add weight routes

### PR DESCRIPTION
**Pull request by danielburnley**

### CODE REVIEW

File: integration_tests/e2e/personal/edit/editWeight.cy.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      cy.setupBannerStubs({ prisonerNumber, bookingId })
+      cy.task('stubPrisonPerson', { prisonerNumber })
+      cy.task('stubPrisonPersonUpdatePhysicalAttributes', { prisonerNumber })
+      cy.setupPersonalPageSubs({ prisonerNumber, bookingId })
+      cy.task('stubPersonalCareNeeds')
+    },
+    editUrl: `prisoner/${prisonerNumber}/personal/edit/weight/imperial`,
+    validInputs: { textInputs: { stones: '5', pounds: '3' } },
+    invalidResponses: [
+      // {
```
Comment (danielburnley) [inline integration_tests/e2e/personal/edit/editWeight.cy.ts:87]: I'm updating the validation in a subsequent PR so i've commented these out for now

File: server/controllers/personalController.ts [hunk 2]
```diff
@@ -175,4 +180,122 @@ export default class PersonalController {
...
     }
   }
+
+  weight() {
+    return {
+      metric: {
+        edit: async (req: Request, res: Response, next: NextFunction) => {
+          const { prisonerNumber } = req.params
+          const { clientToken } = req.middleware
+          const prisonerData: Prisoner = req.middleware?.prisonerData
```
Comment (whitfield-mj) [inline server/controllers/personalController.ts:190]: Add prisonerData to ln 189

File: server/controllers/personalController.ts [hunk 3]
```diff
@@ -175,4 +180,122 @@ export default class PersonalController {
...
+          const { prisonerNumber } = req.params
+          const { clientToken } = req.middleware
+          const { editField } = req.body
+          const errors: HmppsError[] = []
+          const prisonPerson = await this.personalPageService.getPrisonPerson(clientToken, prisonerNumber, true)
+
+          const weight = parseInt(editField, 10)
+          if (Number.isNaN(weight) || weight <= 0) {
+            errors.push({ text: 'Enter a number greater than 0' })
+          }
```
Comment (whitfield-mj) [inline server/controllers/personalController.ts:217]: Didn't we decide not to instantiate and mutate variable?
I would recommend using this as a guard and returning from here then you can remove the if below.

Something like
```javascript
if (Number.isNaN(weight) || weight <= 0) {
     req.flash('errors', [{ text: 'Enter a number greater than 0' }])
     return res.redirect(`/prisoner/${prisonerNumber}/personal/edit/weight`)
 }

try {
  ...
```

File: server/controllers/personalController.ts [hunk 4]
```diff
@@ -175,4 +180,122 @@ export default class PersonalController {
...
+          req.flash('errors', errors)
+          return res.redirect(`/prisoner/${prisonerNumber}/personal/edit/weight`)
+        },
+      },
+
+      imperial: {
+        edit: async (req: Request, res: Response, next: NextFunction) => {
+          const { prisonerNumber } = req.params
+          const { clientToken } = req.middleware
+          const prisonerData: Prisoner = req.middleware?.prisonerData
```
Comment (whitfield-mj) [inline server/controllers/personalController.ts:243]: as above

File: server/controllers/personalController.ts [hunk 5]
```diff
@@ -175,4 +180,122 @@ export default class PersonalController {
...
+          const { stones: stonesString, pounds: poundsString }: { stones: string; pounds: string } = req.body
+          const errors: HmppsError[] = []
+          const prisonPerson = await this.personalPageService.getPrisonPerson(clientToken, prisonerNumber, true)
+
+          const stones = parseInt(stonesString, 10)
+          const pounds = parseInt(poundsString, 10)
+
+          if (Number.isNaN(stones) || stones <= 0 || Number.isNaN(pounds) || pounds <= 0) {
+            errors.push({ text: 'Enter a number greater than 0' })
+          }
```
Comment (whitfield-mj) [inline server/controllers/personalController.ts:276]: as above re returning from the guard

**Review by whitfield-mj — APPROVED** [review]

**Pull request merged**